transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/AlarmManagerScheduler.java

```diff
@@ -95,7 +95,7 @@ public void schedule(TransportContext transportContext, int attemptNumber) {
     Long backendTime = eventStore.getNextCallTime(transportContext);
 
     long timeDiff = 0;
-    if (backendTime != null) {
+    if (backendTime != 0) {
```

Comment:
Is this check even needed? Would it make sense to replace with a single expression like: `long timeDiff = Math.max(0, backendTime - clock.getTime())`?